src/main/java/com/github/manolo8/darkbot/core/objects/facades/InventoryProxy.java

```diff
@@ -0,0 +1,60 @@
+package com.github.manolo8.darkbot.core.objects.facades;
+
+import com.github.manolo8.darkbot.Main;
+import com.github.manolo8.darkbot.core.itf.Updatable;
+import com.github.manolo8.darkbot.core.objects.swf.ObjArray;
+import eu.darkbot.api.managers.InventoryAPI;
+import eu.darkbot.util.Timer;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.ToString;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.github.manolo8.darkbot.Main.API;
+
+@Getter
+public class InventoryProxy extends Updatable implements InventoryAPI {
+    @Getter(AccessLevel.NONE)
+    private final ObjArray itemsArr = ObjArray.ofVector(true);
+
+    public List<Item> items = new ArrayList<>();
+    private final Timer updateTimer = Timer.get(10_000);
+
+    @Override
+    public void update() {
+        if (updateTimer.tryActivate()) {
+            itemsArr.update(Main.API.readMemoryPtr(address, 0x30, 0x58));
+            itemsArr.syncAndReport(items, Item::new);
```

Comment:
Sync and report doesn't seem to be used for anything, you don't care if an update was made or not, so why use it?